index.bs

```diff
@@ -2201,7 +2201,7 @@ partial interface MLGraphBuilder {
         - *options*: an optional {{MLLstmCellOptions}}. The optional parameters of the operation.
             - *bias*: an {{MLOperand}}. The 1-D input bias tensor of shape [4 * hidden_size]. The ordering of the bias vectors in the first dimension of the tensor shape is specified according to the *options.layout* argument.
             - *recurrentBias*: an {{MLOperand}}. The 1-D recurrent bias tensor of shape [4 * hidden_size]. The ordering of the bias vectors in the first dimension of the tensor shape is specified according to the *options.layout* argument.
-            - *peepholeWeight*: an {{MLOperand}}. The 1-D weight tensor for peepholes of shape [3 * hidden_size]. The pack ordering of the weight vectors is for the *input (i)*, *output (o)*, and *forget (f)* gate respectively.
+            - *peepholeWeight*: an {{MLOperand}}. The 1-D weight tensor for peepholes of shape [4 * hidden_size]. The pack ordering of the weight vectors is for the *input (i)*, *output (o)*, and *forget (f)* gate respectively.
```

Comment:
So I'm not sure about TF and PT (I can't figure out yet which one corresponds to the "peephole", since they evidently use a different term), but the DML API definitely shows `PeepholeTensor sizes = { 1, 1, num_directions, 3 * hidden_size }`, not 4?

- https://www.tensorflow.org/api_docs/python/tf/keras/layers/LSTM
- https://pytorch.org/docs/stable/generated/torch.nn.LSTM.html
- https://learn.microsoft.com/en-us/windows/win32/api/directml/ns-directml-dml_lstm_operator_desc

Reply:
Thanks @fdwr!

In the Spec, `lstm` operation describes **peepholeWeight** as bellowing:
```
peepholeWeight: an [MLOperand](https://webmachinelearning.github.io/webnn/#mloperand). 
The 2-D weight tensor for peepholes of shape [num_directions, 4 * hidden_size]. 
The pack ordering of the weight vectors is for the input (i), output (o), and forget (f) gate respectively.
```  
So I did this modification for **lstmCell** operation to align with the second value of above shape **[num_directions, 4 * hidden_size]**. 

> but the DML API definitely shows PeepholeTensor sizes = { 1, 1, num_directions, 3 * hidden_size } and ONNX too P (optional, differentiable) : The weight tensor for peepholes. ... It has shape [num_directions, 3*hidden_size]

I verified that **[3 * hidden_size]** couldn't be the shape of **peepholeWeight** for `lstm` operation with the error when executing slice compute, while the shape of **peepholeWeight** for `lstmCell` operation being **[3 * hidden_size]** works.

Reply:
Oof, you had to dig back into the original paper. Thank you for the spec wording and pseudocode fix.